[Bug] PAX build error on Rocky Linux 8

### Apache Cloudberry version

main

### What happened

```
build pax, USE_MANIFEST_API=OFF USE_PAX_CATALOG=ON
-- The C compiler identification is GNU 8.5.0
-- The CXX compiler identification is GNU 8.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- pg_config.h => /home/gpadmin/cloudberry/contrib/pax_storage/../../src/include/pg_config.h
-- Looking for USE_ASSERT_CHECKING
-- Looking for USE_ASSERT_CHECKING - not found
-- Looking for USE_LZ4
-- Looking for USE_LZ4 - found
-- PAX BUILD TYPE: Release
-- Looking for USE_VECTORIZATION
-- Looking for USE_VECTORIZATION - not found
-- Build pax without vectorization support
-- Found Protobuf: /usr/lib64/libprotobuf.so;-lpthread (found version "3.5.0")
-- Found Protobuf: /usr/lib64/libprotobuf.so;-lpthread (found suitable version "3.5.0", minimum required is "3.5.0")
-- Checking for module 'libzstd'
--   Found libzstd, version 1.4.4
-- tabulate version: v1.5.0
-- Using C++11
-- Configuring done (0.7s)
-- Generating done (0.0s)
-- Build files have been written to: /home/gpadmin/cloudberry/contrib/pax_storage/build
cd build && make -j8
make[2]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: warning: -j8 forced in submake: resetting jobserver mode.
make[3]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  0%] Built target generate_protobuf
[  1%] Building C object src/cpp/CMakeFiles/generate_sql_script_program.dir/__/__/tools/gen_sql.c.o
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[4]: Entering directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/orc_proto.pb.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/micro_partition_stats.pb.cc.o
[  2%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/proto/pax.pb.cc.o
[  3%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column.cc.o
[  4%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column_traits.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_columns.cc.o
[  5%] Building CXX object src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_compress.cc.o
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:76: src/cpp/CMakeFiles/paxformat.dir/storage/proto/micro_partition_stats.pb.cc.o] Error 1
make[4]: *** Waiting for unfinished jobs....
cc1plus: error: -Werror=pessimizing-move: no option -Wpessimizing-move
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:160: src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_columns.cc.o] Error 1
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:104: src/cpp/CMakeFiles/paxformat.dir/storage/proto/orc_proto.pb.cc.o] Error 1
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:90: src/cpp/CMakeFiles/paxformat.dir/storage/proto/pax.pb.cc.o] Error 1
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:132: src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column.cc.o] Error 1
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:146: src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_compress.cc.o] Error 1
make[4]: *** [src/cpp/CMakeFiles/paxformat.dir/build.make:118: src/cpp/CMakeFiles/paxformat.dir/storage/columns/pax_column_traits.cc.o] Error 1
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[3]: *** [CMakeFiles/Makefile2:149: src/cpp/CMakeFiles/paxformat.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
[  5%] Linking C executable generate_sql_script_program
make[4]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
[  5%] Built target generate_sql_script_program
make[3]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[2]: *** [Makefile:136: all] Error 2
make[2]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage/build'
make[1]: *** [Makefile:69: build] Error 2
make[1]: Leaving directory '/home/gpadmin/cloudberry/contrib/pax_storage'
make: *** [GNUmakefile:32: all] Error 2
make: Leaving directory '/home/gpadmin/cloudberry'
```

OS + gcc/g++:

```
[gpadmin@cdw cloudberry]$ cat /etc/os-release
NAME="Rocky Linux"
VERSION="8.10 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.10"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.10 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2029-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.10"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.10"


[gpadmin@cdw cloudberry]$ gcc --version
gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-28)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[gpadmin@cdw cloudberry]$ g++ --version
g++ (GCC) 8.5.0 20210514 (Red Hat 8.5.0-28)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### What you think should happen instead

_No response_

### How to reproduce

Following the build docs: https://cloudberry.apache.org/docs/next/deployment/quick-build

### Operating System

Rocky Linux 8.10

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

--- a/contrib/pax_storage/CMakeLists.txt
+++ b/contrib/pax_storage/CMakeLists.txt
@@ -21,7 +21,10 @@ set(CMAKE_CXX_STANDARD 17)
 set(TOP_DIR ${PROJECT_SOURCE_DIR}/../..)
 set(CBDB_INCLUDE_DIR ${TOP_DIR}/src/include)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Werror=pessimizing-move  -Wno-unused-function -Wno-error=ignored-qualifiers -Wno-error=array-bounds  -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-parameter -Wno-clobbered -Wno-sized-deallocation -g")
+# Base CXX flags
+# Note: -Wpessimizing-move is enabled by default in GCC 9+ and will be caught by -Werror
+# No need to explicitly add -Werror=pessimizing-move (which breaks GCC 8.x compatibility)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wno-unused-function -Wno-error=ignored-qualifiers -Wno-error=array-bounds -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-parameter -Wno-clobbered -Wno-sized-deallocation -g")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-unused-parameter -Wno-parameter-name")
 
 option(USE_MANIFEST_API "Use manifest API" OFF)

--- a/contrib/pax_storage/src/cpp/catalog/pax_aux_table.cc
+++ b/contrib/pax_storage/src/cpp/catalog/pax_aux_table.cc
@@ -700,7 +700,7 @@ pax::MicroPartitionMetadata PaxGetMicroPartitionMetadata(Relation rel,
     paxc::FetchMicroPartitionAuxRow(rel, snapshot, block_id,
                                     FetchMicroPartitionAuxRowCallbackWrapper,
                                     &ctx);
-    return std::move(ctx.info);
+    return ctx.info;
   }
   CBDB_WRAP_END;
 }

--- a/contrib/pax_storage/src/cpp/comm/fast_io.cc
+++ b/contrib/pax_storage/src/cpp/comm/fast_io.cc
@@ -27,6 +27,17 @@
 
 #include "fast_io.h"
 
+#include <unistd.h>  // for pread
+
+// uring_likely may not be defined in older liburing versions
+#ifndef uring_likely
+#if __GNUC__ >= 3
+#define uring_likely(x) __builtin_expect((x) != 0, 1)
+#else
+#define uring_likely(x) ((x) != 0)
+#endif
+#endif
+
 namespace pax
 {
 

